server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -2399,7 +2399,7 @@ private ClusterState getClusterStateWithUpdatedRepositoryGeneration(ClusterState
         return ClusterState.builder(currentState)
             .metadata(
                 Metadata.builder(currentState.getMetadata())
-                    .putCustom(
+                    .putDefaultProjectCustom(
```

Comment:
This makes the repository explicitly work with the default project. Otherwise, it throws MultiProjectPendingException in some multi-project enabled tests. It worked previously because repositories are cluster custom which does not check the number of projects. It is a temporary workaround (the method is marked as deprecated for removal) and will be removed once repository and snapshot fully support multi-project.